content/articles/dealing-with-python's-missing-values-a-complete-guide/index.md

```diff
@@ -0,0 +1,105 @@
+2. Qualitative or feature variables are also included in the list of quantitative and objective variables.
+### The Different Types of Imputed Information
+A variety of sizes and shapes are offered in the form of imputations. In order to build an accurate model of our application, we must first fill in any data gaps that exist in our dataset. This is one of the techniques for doing so.
+- The term univariate or mean imputation is used when only the target variable is imputed.
+- Linear regression can be used to predict missing values based on other variables, or multivariate imputation can be used to predict missing values based on a single variable.
+- All missing values in the dataset can be imputed once to create an imputed set of values.
+- Several imputations are needed if the same missing values are imputed in different places in the same dataset more than once. This simply requires repeating a single imputation to produce a large number of imputed datasets from a single source dataset.
+### How to fix our dataset's missing data
+There are a variety of approaches to dealing with missing data; we will look at some of them, but first, we will start from the beginning with things like importing libraries.
+```Python
+import pandas as pan
+import numpy as num
+
+dataset = pan.read_csv("IncomeAndGender.csv")
+dataset.head()
+```
+Looking at the dataset's dimensions as a measure of its size:
+```Python
+dataset.shape
+```
+In the search for missing information:
+```Python
+print(dataset.isnull().sum())
+```
+OUTPUT
+```Bash
+Salary          0
+Gender          0
+AgeNumber       16
+PhD             13
+dType:          int64
+```
+Don't worry about the info that isn't there. It's up to the algorithm to decide how it interprets the information you give it. Missing data affects algorithms in different ways. The optimal imputation values for missing data can be obtained by training loss reduction algorithms.
+
+Using linear regression as an example, an error can be made. As soon as we know what went wrong with the model, we have to figure out how to deal with the lack of data until we get a clearer grasp of what went wrong in the first place. If the reaction is positive, we'll proceed. If it is negative, we'll stop.
+```Python
+dataset["AgeNumber"][:10]
+```
+### Removing the rows/columns that are not in use
+The next simplest strategy is to exclude observations with missing data. In the end, you can miss out on important pieces of information. Python's pandas module has a dropna() method for getting rid of empty columns. When dealing with machine learning issues, rely on your own experience or that of a domain expert rather than attempting to fill in every blank in every column.
+
+There are both advantages and disadvantages to removing the rows/columns; for example, the model becomes more robust after deleting missing data. If you have a large amount of missing data, your model will be less accurate.

```

Comment:
`There are both advantages and disadvantages to removing the rows/columns`

Please state them